sys/posix/sockets/posix_sockets.c

```diff
@@ -363,8 +361,8 @@ int accept(int socket, struct sockaddr *restrict address,
                     res = -1;
                     break;
                 }
-                _htons_port(port);  /* XXX: sin(6)_port is supposed to be network byte
-                                     *      order */
+                *port = htons(*port); /* XXX: sin(6)_port is supposed to be
```

Comment:
Then use `byteorder_htons()`, please to be type-safe (sorry for post-ACK commenting, just stumbled upon this)